lib/Differentiator/DiffPlanner.cpp

```diff
@@ -620,11 +620,20 @@ namespace clad {
     if (!EnableVariedAnalysis)
       return true;
 
-    if (VD->getType()->isPointerType() || isa<ArrayType>(VD->getType()))
-      return true;
-
     if (!m_ActivityRunInfo.HasAnalysisRun) {
-      std::copy(Function->param_begin(), Function->param_end(),
+      ArrayRef<ParmVarDecl*> FDparam = Function->parameters();
+      std::vector<ParmVarDecl*> derivedParam;
+
+      for (auto* parameter : FDparam) {
+        QualType parType = parameter->getType();
+        if (parType->isPointerType()) {
+          if (!parType->getPointeeType().isConstQualified())
+            derivedParam.push_back(parameter);
+        } else if (!parType.isConstQualified())
+          derivedParam.push_back(parameter);
+      }
```

Comment:
```suggestion
      for (ParmVarDecl* Param : FDparam) {
        // Ignore const qualified types.
        QualType ParamTy = Param->getType();
        do {
           ParamTy = ParamTy->getPointeeType();
        } while(ParamTy->isPointerType());
        if (!ParamTy.isConstQualified())
          derivedParam.push_back(Param);
      }
```
Should cover cases such as `const*const**`.